02-set-up.Rmd

```diff
@@ -125,32 +109,33 @@ server management. Fortunately there are many tools designed to ease monitoring
 - On Linux, the shell command `top` displays key resource use figures for most
 distributions. `htop` and Gnome's **System Monitor** (`gnome-system-monitor`, see figure \@ref(fig:2-1)) are more refined alternatives which use command-line and graphical user interfaces respectively.
 A number of options such as `nethogs` monitor internet usage.
-- On Windows the **Task Manager** provides key information on RAM and CPU use by
-process. This can be started in modern Windows versions by typing `Ctrl-Alt-Del` or by clicking the task bar and 'Start Task Manager'.
 - On Mac the **Activity Monitor** provides similar functionality. This can be initiated
 from the Utilities folder in Launchpad.
+- On Windows the **Task Manager** provides key information on RAM and CPU use by
+process. This can be started in modern Windows versions by typing `Ctrl-Alt-Del` or by clicking the task bar and 'Start Task Manager'.
 
 #### Exercises {-}
 
 1. What is the exact version of your computer's operating system?
-2. Start an activity monitor then type and execute the following code. How do the results on your system compare to those presented in Figure \@ref(fig:2-1)? 
+2. Start an activity monitor then type and execute the following code.
+In it `lapply()` (or parallel version `parLapply()`) is used to *apply* a function, `median()` over every element in a list (see Section \@ref(the-apply-family)). How do the system output logs results on your system compare to those presented in Figure \@ref(fig:2-1)? 
 
     ```{r, eval=FALSE}
-    # Note: this will consume more than 2 GB RAM and may take several seconds to run depending on hardware
+    # Note: uses 2+ GB RAM and several seconds or more depending on hardware
     # 1: Create large dataset
     X = as.data.frame(matrix(rnorm(1e8), nrow = 1e7))
     # 2: Find the median of each column using a single core
     r1 = lapply(X, median)
     # 3: Find the median of each column using many cores
-    r2 = parallel::mclapply(X, median) # runs in serial on Windows
+    r2 = parallel::parLapply(X, median) 
```

Comment:
This doesn't work; you need to start a cluster.  I would move this example to the chapter/section on parallel programming.